cart-service/src/main/java/com/petclinic/cartsservice/businesslayer/CartServiceImpl.java

```diff
@@ -37,4 +39,27 @@ public Mono<CartResponseModel> getCartByCartId(String cartId) {
                             .map(products -> EntityModelUtil.toCartResponseModel(cart, products));
                 });
     }
+
+    @Override
+    public Mono<CartResponseModel> updateCartByCartId(Mono<CartRequestModel> cartRequestModel, String cartId) {
+        return cartRepository.findCartByCartId(cartId)
+                .switchIfEmpty(Mono.defer(() -> Mono.error(new NotFoundException("Cart id was not found: " + cartId))))
+                .flatMap(foundCart -> cartRequestModel
+                        .map(EntityModelUtil::toCartEntity)
+                        .doOnNext(c -> c.setCartId(foundCart.getCartId()))
+                        .doOnNext(c -> c.setCustomerId(foundCart.getCustomerId()))
+                        .doOnNext(c -> c.setProductIds(foundCart.getProductIds()))
+                        .doOnNext(c -> c.setId(foundCart.getId()))
+                )
+
```

Comment:
what you do here is :
1. find the actual cart
2. map the request to an entity
3. take the data from the found entity and put it in the request that you have transformed into an entity
this means that the update is not going to work